lib/AST/ASTPrinter.cpp

```diff
@@ -859,11 +859,16 @@ class PrintAST : public ASTVisitor<PrintAST> {
 
       // Get the innermost nominal type context.
       DeclContext *DC;
-      if (isa<NominalTypeDecl>(Current) || isa<ExtensionDecl>(Current))
+      if (isa<NominalTypeDecl>(Current))
         DC = Current->getInnermostDeclContext();
+      else if (isa<ExtensionDecl>(Current))
+        DC = Current->getInnermostDeclContext()->
+          getAsNominalTypeOrNominalTypeExtensionContext();
       else
         DC = Current->getDeclContext();
 
+      assert(DC->isTypeContext());
```

Comment:
Basically what I don't understand is how getContextSubstitutions(ExtensionDecl) is different from getContextSubstitutions(ExtensionDecl->getAsNominalTypeOrNominalTypeExtension())